### PR TITLE
[ADF-1880] More configuration options for adf-start-process component

### PR DESCRIPTION
**Pull request by eromano**
**Please check if the PR fulfills these requirements**

> - [X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [X] Tests for the changes have been added (for bug fixes / features)
> - [X] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ADF-2190
https://issues.alfresco.com/jira/browse/ADF-2191
https://issues.alfresco.com/jira/browse/ADF-1880
https://issues.alfresco.com/jira/browse/ADF-2164

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/process-services/process-list/components/start-process.component.ts [hunk 1]
```diff
@@ -96,37 +96,41 @@ export class StartProcessInstanceComponent implements OnChanges {
             this.appId = changes['appId'].currentValue;
         }
 
+        if (changes['processDefinitionId'] && changes['processDefinitionId'].currentValue) {
```
Comment (popovicsandras) [inline lib/process-services/process-list/components/start-process.component.ts]: Why do you need this? processDefinitionId being an input parameter itself, I suppose it should be updated the same way in the background by angular. Am I wrong?

File: lib/process-services/process-list/components/start-process.component.ts [hunk 2]
```diff
@@ -96,37 +96,41 @@ export class StartProcessInstanceComponent implements OnChanges {
...
-                            this.processDefinitions = this.processDefinitions.filter((currentProcessDefinition) => {
-                                return currentProcessDefinition.id === this.processDefinitionId;
-                            });
-                            this.currentProcessDef = JSON.parse(JSON.stringify(this.processDefinitions[0]));
-                        }
+        this.activitiProcess.getProcessDefinitions(this.appId).subscribe(
+            (processDefinitionRepresentations: ProcessDefinitionRepresentation[]) => {
+                this.processDefinitions = processDefinitionRepresentations;
+
+                if (this.processDefinitions.length === 1 || !this.processDefinitionId) {
```
Comment (mauriziovitale) [inline lib/process-services/process-list/components/start-process.component.ts]: what happens if I have multiple process definitions and I didn't use the processDefId ?
This condition will be 
false || true

File: lib/process-services/process-list/components/start-process.component.ts [hunk 3]
```diff
@@ -96,37 +96,41 @@ export class StartProcessInstanceComponent implements OnChanges {
...
+                    this.selectedProcessDef = this.processDefinitions[0];
+                } else {
+                    this.selectedProcessDef = this.processDefinitions.find((currentProcessDefinition) => {
+                        return currentProcessDefinition.id === this.processDefinitionId;
+                    });
+
+                    if (!this.selectedProcessDef) {
```
Comment (mauriziovitale) [inline lib/process-services/process-list/components/start-process.component.ts]: Are we saying that if I didn't found the processDefinitionId inside the list, the first one will be selected?

File: lib/process-services/process-list/components/start-process.component.ts (hunk 1 above)
Comment (mauriziovitale) [reply lib/process-services/process-list/components/start-process.component.ts]: In addition, the input is not processDefinitionId but **processDefinition**

**Pull request merged**